app/models/annotation.rb

```diff
@@ -72,6 +72,8 @@ def check_if_released
 
     return if result.is_a_review? && !result.released_to_students
 
+    return if result.submission.non_pr_results.where('results.released_to_students': true).empty?
```

Comment:
I'm not convinced another check is necessary. Can't we modify line 65 to use `non_pr_results`?

Reply:
@david-yz-liu Thanks! I think you are correct, and in fact this logic is better.